nextmv/nextmv/local/runner.py

```diff
@@ -0,0 +1,312 @@
+    ----------
+    app_id : str
+        The ID of the application.
+    src : str
+        The path to the application source code.
+    manifest : Manifest
+        The application manifest.
+    run_config : dict[str, Any]
+        The run configuration.
+    name : Optional[str], optional
+        The name for the run, by default None.
+    description : Optional[str], optional
+        The description for the run, by default None.
+    input_data : Optional[Union[dict[str, Any], str]], optional
+        The input data for the run, by default None. If `inputs_dir_path` is
+        provided, this parameter is ignored.
+    inputs_dir_path : Optional[str], optional
+        The path to the directory containing input files, by default None. If
+        provided, this parameter takes precedence over `input_data`.
+    options : Optional[dict[str, Any]], optional
+        Additional options for the run, by default None.
+
+    Returns
+    -------
+    str
+        The ID of the created run.
+    """
+
+    # Check for required optional dependencies
+    missing_deps = []
+    if importlib.util.find_spec("folium") is None:
+        missing_deps.append("folium")
+    if importlib.util.find_spec("plotly") is None:
+        missing_deps.append("plotly")
+
+    if missing_deps:
+        raise ImportError(
+            f"{' and '.join(missing_deps)} {'is' if len(missing_deps) == 1 else 'are'} not installed. "
+            "Please install optional dependencies with `pip install nextmv[all]`"
+        )
+
+    # Initialize the run: create the ID, dir, and write the input.
+    run_id = safe_id("local")
+    run_dir = new_run(
+        app_id=app_id,
+        src=src,
+        run_id=run_id,
+        run_config=run_config,
+        name=name,
+        description=description,
+    )
+    record_input(
+        run_dir=run_dir,
+        run_id=run_id,
+        input_data=input_data,
+        inputs_dir_path=inputs_dir_path,
+    )
+
+    # Start the process as a daemon (detached) so we don't wait for it to
+    # finish. We send the input via stdin and close it immediately without
+    # waiting. We call the `executor.py` script to do the actual execution.
+    stdin_input = json.dumps(
+        {
+            "run_id": run_id,
+            "src": os.path.abspath(src),
+            "manifest_entrypoint": manifest.entrypoint,
+            "run_dir": os.path.abspath(run_dir),
+            "run_config": run_config,
+            "input_data": input_data,
+            "inputs_dir_path": os.path.abspath(inputs_dir_path) if inputs_dir_path is not None else None,
+            "options": options,
+        }
+    )
+    args = [sys.executable, "executor.py"]
+    process = subprocess.Popen(
+        args,
+        env=os.environ,
+        text=True,
+        stdin=subprocess.PIPE,
+        stdout=subprocess.DEVNULL,
+        stderr=subprocess.DEVNULL,
+        cwd=os.path.dirname(__file__),
+        start_new_session=True,  # Detach from parent process
```

Comment:
Is there a way to interact with an ongoing run(s)? Maybe not right now, but in future? :thinking: 
I can easily imagine spawning a number of runs (maybe even accidentally) and then not knowing when they are done, what their status is, etc.